core/src/main/scala/org/apache/spark/SecurityManager.scala

```diff
@@ -187,6 +187,7 @@ private[spark] class SecurityManager(
     val ioEncryptionKey: Option[Array[Byte]] = None)
   extends Logging with SecretKeyHolder {
 
+  def this(sparkConf: SparkConf) {this(sparkConf, None)}
```

Comment:
Add a blank line after. I think this is more conventional as

`def this(sparkConf: SparkConf) = this(sparkConf, None)`

Because this is private to Spark, is it really something you need to extend?
That is, is there a good case to make this an SPI of sorts?